backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java

```diff
@@ -124,7 +147,169 @@ public String caseDefinition(Definition def) {
+            if (subsettedFeature != null) {
+                return subsettedFeature;
+            } else {
+                return subsettedFeature;
+            }
+        };
+    }
+
+    private void appendRedefinition(Appender builder, List<Redefinition> redefinitions, Element element) {
+        if (!redefinitions.isEmpty()) {
+            builder.appendSpaceIfNeeded().append(LabelConstants.REDEFINES);
+            builder.appendSpaceIfNeeded().append(redefinitions.stream().map(Redefinition::getRedefinedFeature)
+                    .filter(Objects::nonNull)
+                    .map(superFeature -> getDeresolvableName(superFeature, element))
+                    .collect(Collectors.joining(", ")));
+        }
+
+    }
+
+    private void appendReferenceSubsetting(Appender builder, ReferenceSubsetting ownedReferenceSubsetting, Element element) {
+        if (ownedReferenceSubsetting != null) {
+            builder.appendSpaceIfNeeded().append(LabelConstants.REFERENCES);
+            if (ownedReferenceSubsetting.getReferencedFeature() != null) {
+                builder.appendSpaceIfNeeded().append(getDeresolvableName(ownedReferenceSubsetting.getReferencedFeature(), element));
+            }
+        }
+
+    }
+
+    /**
+     * @param builder
+     * @param ownedTyping
+     * @param element
+     */
+    private void appendFeatureTyping(Appender builder, EList<FeatureTyping> ownedTyping, Element element) {
+        if (!ownedTyping.isEmpty()) {
+            builder.appendSpaceIfNeeded().append(LabelConstants.DEFINED_BY);
+            builder.appendSpaceIfNeeded().append(ownedTyping.stream()
+                    .filter(ft -> ft.getType() != null)
+                    .map(ft -> Tuples.of(ft, ft.getType()))
+                    .filter(t -> t.getT2() != null)
+                    .map(t -> {
+                        String name = getDeresolvableName(t.getT2(), element);
+
+                        if (t.getT1() instanceof ConjugatedPortTyping) {
+                            name = LabelConstants.CONJUGATED + name;
+                        }
+                        return name;
+                    }
+
+                    )
+                    .collect(Collectors.joining(", ")));
+        }
+    }
+
+    private void appendMultiplicityPart(Appender builder, Feature feature) {
+        MultiplicityRange multiplicity = feature.getOwnedElement().stream()
+                .filter(MultiplicityRange.class::isInstance)
+                .map(MultiplicityRange.class::cast)
+                .findFirst()
+                .orElse(null);
+
+        if (multiplicity != null) {
+            String expression = multiplicity.getOwnedElement().stream()
+                    .filter(element -> element instanceof LiteralExpression || element instanceof FeatureReferenceExpression)
+                    .map(element -> {
+                        String exp = null;
+                        if (element instanceof LiteralExpression lit) {
```

Comment:
I think it would be easir to use _getLowerBound_ and _getUpperBound_ methods here it would make the code easier and more robust (you need to implement them in the metamodel code).

Then for each non empty bound : ```expression = doSwitch(range.getUpper/LowerBound())``

Reply:
done